### DISCUSSION TITLE
FR: Override / access code - configurable random

### DISCUSSION THREAD

**Discussion by woj-tek** [proginosko/LeechBlockNG#548] [Ideas]
Hi,
would it be possible to add two options to override options:
1) variable random string lenght (currently only 32, 64 and 128)
2) set of characters (A-Z, a-z, 0-9, special characters)

I'm aware that it's supposed to be difficult but with currently generate password I tend to disable LB altogether to bypass the limits, but it also removes any timers that would prompt me to stop.

This is somewhat inspired by KeePassXC password generator: 
![](https://keepassxc.org/images/screenshots/password_generator_advanced.png)

**Comment by woj-tek** [comment]
Hmm... this was moved to "discussions". Would it be possible to add this as an option? Or even more - have it completely configurable like in KeePassXC generator (length/types or use "phrase")?

<img width="778" alt="Captura de pantalla 2025-03-01 a las 16 36 26" src="https://github.com/user-attachments/assets/969f7f09-7ba0-4ac6-a800-e79b387e60dc" />
<img width="764" alt="Captura de pantalla 2025-03-01 a las 16 36 34" src="https://github.com/user-attachments/assets/3e043c9d-0083-4397-8222-dbe8aa6c4cd1" />

**Comment by slepto** [comment]
I don't quite follow what you mean by this, "I'm aware that it's supposed to be difficult but with currently generate password I tend to disable LB altogether to bypass the limits, but it also removes any timers that would prompt me to stop."

but I could point out that (at least if you're a mac user) you can use a policy.json file to force-enable leechblock in Firefox.  Also remove guest mode and incognito mode because those are workarounds.  Use an admin account on your mac to set up the policy and a standard account for your computer use.  

I do this and I can no longer turn leechblock off.  Forces me to leverage overrides, lockdowns, and the occasional re-administering of options inside access control.

**Reply by woj-tek** [reply]
> I don't quite follow what you mean by this, "I'm aware that it's supposed to be difficult but with currently generate password I tend to disable LB altogether to bypass the limits, but it also removes any timers that would prompt me to stop."

I avoided doing that because I quite often had some "to read" tabs that sometimes got timed out so disabling LB would evaporate those hence... for some reason I don't tend to avoid LB (was doing it before though)
